hail/python/hail/expr/functions.py

```diff
@@ -1827,6 +1827,64 @@ def rand_norm(mean=0, sd=1, seed=None) -> Float64Expression:
     return _seeded_func("rand_norm", tfloat64, seed, mean, sd)
 
 
+@typecheck(mean=nullable(expr_array(expr_float64)), cov=nullable(expr_array(expr_float64)), seed=nullable(int))
+def rand_norm2d(mean=None, cov=None, seed=None) -> ArrayNumericExpression:
+    """Samples from a normal distribution with mean `mean` and covariance matrix `cov`.
+
+    Examples
+    --------
+
+    >>> hl.eval(hl.rand_norm2d())  # doctest: +NOTEST
+    [0.5515477294463427, -1.1782691532205807]
+
+    >>> hl.eval(hl.rand_norm2d())  # doctest: +NOTEST
+    [-1.127240906867922, 1.4495317887283203]
+
+    Notes
+    -----
+    The covariance of a 2d normal distribution is a 2x2 symmetric matrix
+    [[a, b], [b, c]]. This is specified in `cov` as a length 3 array [a, b, c].
+    The covariance matrix must be positive semi-definite, i.e. a>0, c>0, and
+    a*d - c^2 > 0.
+
+    If `mean` and `cov` are both None, draws from the standard 2d normal
+    distribution.
+
+    Parameters
+    ----------
+    mean : :class:`.ArrayNumericExpression`, optional
+        Mean of normal distribution. Array of length 2.
+    cov : :class:`.ArrayNumericExpression`, optional
+        Covariance of normal distribution. Array of length 3.
+    seed : :obj:`int`, optional
+        Random seed.
+
+    Returns
+    -------
+    :class:`.ArrayFloat64Expression`
+    """
+    if mean is None:
+        m1 = 0
+        m2 = 0
+    else:
+        m1 = mean[0]
```

Comment:
we'll need to wrap this in a `bind` so that we don't evaluate the arguments multiple times.

Something like:

```
if mean is None:
    mean = [0, 0]
if cov is None:
    cov = [1, 0, 1]

def f(mean, cov):
    m1 = mean[0]
    ... 

return hl.rbind(mean, cov, f)
```